src/screens/map-screen/map-screen.component.tsx

```diff
@@ -189,11 +201,67 @@ const MapScreen = () => {
    */
   useEffect(() => {
     setCurrentRegion(getCampusById(CampusId.SGW).region);
+    if (!currentLocation) {
+      getCurrentLocationAsync().then(location => {
+        setCurrentLocation({
+          latitude: location.coords.latitude,
+          longitude: location.coords.latitude
+        });
+      });
+    }
   }, []);
 
+  /**
+   *  Set the value of destination or initial locaiton depending
+   *  on which input the user is focued
+   * @param poi
+   */
+  const setMarkerLocation = (poi: POI | null) => {
+    if (markerSetsDestination) {
+      setDestination(poi);
+    } else {
+      setInitialLocation(poi);
+    }
+  };
+
+  /**
+   * Filters an array of POIs based on users input
+   * @param userInput
+   */
+  const queryText = (userInput, setAutocomplete, onChangeText) => {
```

Comment:
This shouldn't be in the in the mapscreen imo. It could be made pure and placed in a service